core/src/main/java/org/apache/iceberg/GenericStatisticsFile.java

```diff
@@ -18,13 +18,14 @@
  */
 package org.apache.iceberg;
 
+import java.io.Serializable;
 import java.util.List;
 import java.util.Objects;
 import java.util.StringJoiner;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
 
-public class GenericStatisticsFile implements StatisticsFile {
+public class GenericStatisticsFile implements StatisticsFile, Serializable {
```

Comment:
Why does this need to be serializable?

Reply:
Oh, I see. For `SerializableTable`. I don't think that we need to actually serialize these. Instead, you can delegate to `lazyTable()` like some of the other methods. We don't expect serialized versions to use this since it is for planning on the coordinator or driver.